examples/gno.land/r/demo/profile/profile.gno

```diff
@@ -0,0 +1,312 @@
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).AvatarURL
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).AvatarURL
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display age from address or username
+func DisplayAge(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Age
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Age
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display gender from address or username
+func DisplayGender(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Gender
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Gender
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display website from address or username
+func DisplayWebsite(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Website
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Website
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display country from address or username
+func DisplayCountry(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).Country
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).Country
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display city from address or username
+func DisplayCity(address std.Address, username string) string {
+	if address != "" {
+		if value, exists := profiles.Get(address.String()); exists {
+			return value.(UserProfile).City
+		}
+	} else if username != "" {
+		if addr, exists := usernameIndex.Get(username); exists {
+			if value, exists := profiles.Get(addr.(string)); exists {
+				return value.(UserProfile).City
+			}
+		}
+	}
+	return "Address or Username not found"
+}
+
+// Display modification history specific to a field from address or username
+func DisplayModificationHistory(address std.Address, username, field string) string {
+	if address != "" {
```

Comment:
Try to simplify these as much as possible - nested if's are not very readable.

Reply:
I've simplified the function, and I'm wondering if it would be necessary to adapt the function so that if the function is called without specifying a field, all the history is returned? [commit](https://github.com/gnolang/gno/pull/1983/commits/d1c108ca8b481ae249567b5b24ed25db7f52a096)